### ISSUE TITLE
Arrows are not laid to the target

### ISSUE THREAD

**Issue by Paul13517** [Neogoma/Navigation-Tutorial-Demo#8]
Hello, thanks for the tutorial, I created a target in a 3d editor and transferred the map in unity, the map transferred normally, 3d objects are shown, but navigation to the target is not built (arrows are not laid). Please tell me how to solve the problem

**Comment by handrianj** [comment]
HI Paul, thanks for reporting.

We need a little bit more details about your issue. When you say that you "transferred the map in unity" are you talking about the unity editor?

Regards

**Comment by Paul13517** [comment]
Hi Handrianj, thanks for the answer!

Yes, about the unity editor.

**Comment by handrianj** [comment]
So in order to display the navrobot you will need to "simulate" a relocation on device.

You can add the following script in your scene:

```csharp
using com.Neogoma.Stardust.API.Relocation;
using com.Neogoma.Stardust.Datamodel;
using UnityEngine;

public class DebuggerLocation : MonoBehaviour
{

    public void SimulateRelocationAtOrigin()
    {
        RelocationResults results = new RelocationResults();
        CoordinateSystem system = new CoordinateSystem();
        MapRelocationManager.Instance.onPositionFound.Invoke(results, system);
        MapRelocationManager.Instance.onLocationFound.Invoke();
    }

    // Update is called once per frame
    public void Update()
    {
        if (Input.GetKeyUp(KeyCode.Space))
        {
            SimulateRelocationAtOrigin();
        }
    }
}
```


This will allow you to press "space" on your computer to simulate a relocation and then the robot should appear.

**Comment by Paul13517** [comment]
if I understood correctly, then this script will also help to build navigation with arrows to the target?

**Comment by handrianj** [comment]
We just updated the project again to remove some faulty components. We didn't run into any issues about updating the target tho, everything works well as long as you place your targets along the mapping path properly as indicated by the [navigation tutorial](https://doc.neogoma.com/#/navigation).

**Comment by Paul13517** [comment]
Thanks for the answer! I studied the documentation carefully. I created a map, opened it in the editor on your site, set the 3d object and the target on the path, imported the map into the unity. When you click on the play, the 3d object is loaded, the robot moves, and for some reason the target always appears at coordinates 0.0.0 where i am didn't install it at all :(

**Comment by handrianj** [comment]
Could you provide us with the map UUID so we have a look at the layout?

**Comment by handrianj** [comment]
With the following flow:

1.  Load the map
2. Click on the navigation button (bottom left)
3. Select the target in the robot canvas (Otherwise "NO TARGET" is selected and navigation won't do anything)
4. Click on "GO" in the robot canvas target

The robot has the expected behavior and goes to the required target.

Can you explain to us how to reproduce your problem step by step ?

**Comment by Paul13517** [comment]
Yes, that's right, I did just that, but after pressing the green button, the target did not appear in the coordinates that I set here https://stardust.neogoma.com/edit/64257a92e8b30
![Снимок экрана 2023-04-01 152006](https://user-images.githubusercontent.com/66047054/229288441-83fb784a-49f5-45b9-8d33-806f37a6cbce.png)
![Снимок экрана 2023-04-01 152241](https://user-images.githubusercontent.com/66047054/229288442-abe0e076-d9ad-4a5b-95f3-2a48b4025eb6.png)

**Comment by Paul13517** [comment]
Sure, 64257a92e8b3001.04.2023, 05:20, "Heri Andrianjafy" ***@***.***>:
Could you provide us with the map UUID so we have a look at the layout?

**Comment by handrianj** [comment]
We just tried with the map you provided and we have no issues with the navigation. Here are our screenshots for reference:

![1](https://user-images.githubusercontent.com/961975/229325198-134580c4-8d5e-4651-8de6-c55bf3149d8a.PNG)
![2](https://user-images.githubusercontent.com/961975/229325200-c5559198-7927-4527-af40-617061912108.PNG)


Do you have any error logs in your console? (please copy paste the stacktrace if you have an error)

**Comment by handrianj** [comment]
Do you have any error logs in your console? If you have can you please copy/paste them here?

**Comment by handrianj** [comment]
We just pushed a new version with simplified logic that we tested on your maps, can you confirm this new version works on your side too ?

If not can you provide error logs if you have any ?

**Comment by Paul13517** [comment]
There are no errors in the console, if step by step: 1) In the map editor, I load the map, change the mod. 2) I create a target within the path. 3) save .4) In the unit I press stardusk, select import cloud and enter ID cards 5) Press play in the unity for simulation. 6) On the UI panel, I select the name of my card. 7) Choose the name of the target. 8) I press the green button GO (my lines are laid to the coordinate 0.0.0 although the coordinates set in the map editor are completely different)

**Comment by handrianj** [comment]
Here's a [video](https://cloud.neogoma.com/s/PJWP2A8y6PFaq5W) of us testing your map based on the current main branch of the repository.

As you can see we manage to make it work (simulating the steps that a user would do on mobile).

**Comment by handrianj** [comment]
Can you confirm you are using the latest version of the repository? (last update was 2 days ago)

**Comment by Paul13517** [comment]
yes, of course, maybe because of the git ignore, i don’t have what you have in the project :D I need to find out what

**Comment by handrianj** [comment]
Do you have the file BotPathRenderer.cs?

**Comment by handrianj** [comment]
Just to be sure you have the BotPathRenderer.cs file right ?

**Comment by handrianj** [comment]
Sure but since you're a paying user feel free to join our discord and you can send us the project in a private channel

https://discord.com/invite/rtJJNEW7JQ

**Comment by handrianj** [comment]
Solved in release 1.14
